html/pfappserver/lib/pfappserver/Base/Form.pm

```diff
@@ -171,6 +176,47 @@ sub id_validator {
    };   
 }
 
+before 'process' => sub {
+    my ($self) = @_;
+    foreach my $field (@{$self->fields}){
+        # Populate all the FingerbankSelect fields with the proper values extracted from the database.
+        if($field->type eq "FingerbankSelect"){
+            # no need for pretty formatting, this is just for validation purposes
+            my @options = map { 
+                {
+                    value => $_->id,
+                    label => $_->id,
+                }
+            } $field->fingerbank_model->all();
+            $field->options(\@options);
+        }
+    }
+};
+
+after 'process' => sub {
```

Comment:
Why not move this logic to pfappserver::Form::Field::FingerbankSelect?
It does not seem like the correct place for this here.